src/containers/pages/InterventionPlan/UpdatePlan/CaseDetails/index.tsx

```diff
@@ -0,0 +1,164 @@
+/** Accordion like component that can be used on the manage plans
+ * page to display case Details
+ */
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import { FlexObject } from 'gisida';
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { toast } from 'react-toastify';
+import { Card, CardBody, CardHeader, Collapse } from 'reactstrap';
+import { ActionCreator, Store } from 'redux';
+import Loading from '../../../../../components/page/Loading';
+import { OPENSRP_API_BASE_URL } from '../../../../../configs/env';
+import {
+  CASE_DETAILS,
+  CASE_INFORMATION,
+  CASE_NUMBER,
+  FAILED_TO_GET_EVENT_ID,
+} from '../../../../../configs/lang';
+import { OPENSRP_FIND_EVENTS_ENDPOINT } from '../../../../../constants';
+import { displayError } from '../../../../../helpers/errors';
+import { growl } from '../../../../../helpers/utils';
+import { OpenSRPService } from '../../../../../services/opensrp';
+import eventReducer, {
+  fetchEvents,
+  FetchEventsAction,
+  getEventById,
+  reducerName,
+} from '../../../../../store/ducks/opensrp/events';
+import {
+  Event,
+  RawEvent,
+  translateEvent,
+  UUID,
+} from '../../../../../store/ducks/opensrp/events/utils';
+import './index.css';
+
+reducerRegistry.register(reducerName, eventReducer);
+
+/** describes props to be passed to caseDetail component */
+export interface CaseDetailsProps {
+  eventId: UUID | null;
+  service: typeof OpenSRPService;
+  event: Event | null;
+  fetchEventsCreator: ActionCreator<FetchEventsAction>;
+}
+
+/** initial props for caseDetails */
+const defaultCaseDetailsProps: CaseDetailsProps = {
+  event: null,
+  eventId: null,
+  fetchEventsCreator: fetchEvents,
+  service: OpenSRPService,
+};
+
+/** async function to load a single event
+ * @param {UUID} eventId - id of the event to get
+ * @param {OpenSRPService} servant - the opensrp service
+ * @param {ActionCreator<FetchEventsAction>} actionCreator - the actionCreator
+ */
+const loadEvent = async (
```

Comment:
Do we need `async` here if we're not `await`ing?

Reply:
will remove that